service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionController.java

```diff
@@ -49,39 +56,31 @@
     private final CaseValidatorService caseValidatorService;
     private final ObjectMapper mapper;
     private final RestrictionsConfiguration restrictionsConfiguration;
+    private final FeeService feeService;
 
-    @Autowired
-    public CaseSubmissionController(
-        UserDetailsService userDetailsService,
-        DocumentGeneratorService documentGeneratorService,
-        UploadDocumentService uploadDocumentService,
-        CaseValidatorService caseValidatorService,
-        ObjectMapper mapper,
-        ApplicationEventPublisher applicationEventPublisher,
-        RestrictionsConfiguration restrictionsConfiguration) {
-        this.userDetailsService = userDetailsService;
-        this.documentGeneratorService = documentGeneratorService;
-        this.uploadDocumentService = uploadDocumentService;
-        this.applicationEventPublisher = applicationEventPublisher;
-        this.caseValidatorService = caseValidatorService;
-        this.mapper = mapper;
-        this.restrictionsConfiguration = restrictionsConfiguration;
-    }
-
+    // TODO: 13/02/2020 update tests
     @PostMapping("/about-to-start")
     public AboutToStartOrSubmitCallbackResponse handleAboutToStartEvent(
         @RequestHeader(value = "authorization") String authorization,
         @RequestBody CallbackRequest callbackRequest) {
+
         CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        List<FeeResponse> fees = feeService.getFees(FeeType.fromOrderType(caseData.getOrders().getOrderType()));
+        BigDecimal amount = feeService.extractFeeToUse(fees)
+            .map(FeeResponse::getAmount)
+            .orElse(BigDecimal.ZERO);
```

Comment:
Is it correct handling? might be worth checking with the business, what IF fees is broken and we can't query? how we should behave?